tools/cabana/tools/search.h

```diff
@@ -0,0 +1,194 @@
+#pragma once
+
+#include <QDialog>
+#include <QPushButton>
+#include <QButtonGroup>
+#include <QVBoxLayout>
+#include <QLineEdit>
+#include <QDoubleValidator>
+#include <QComboBox>
+#include <QSpinBox>
+#include <QLabel>
+#include <QTableWidget>
+#include <QTableWidgetItem>
+
+#include <iostream>
+
+#include "tools/cabana/dbc/dbcmanager.h"
+#include "tools/cabana/streams/abstractstream.h"
+
+uint64_t getBitValue(uint64_t val, int offset, int size);
+
+enum ScanType {
+  ExactValue,
+  BiggerThan,
+  SmallerThan,
+  ValueBetween,
+  IncreasedValue,
+  IncreaseValueBy,
+  DecreasedValue,
+  DecreasedValueBy,
+  ChangedValue,
+  UnchangedValue,
+  UnknownInitialValue
+};
+
+class Sig {
+    public:
+        Sig(MessageId _messageID, int _offset, int _size) : messageID(_messageID), offset(_offset), size(_size) {}
+
+        MessageId messageID;
+        size_t offset;
+        size_t size;
+
+        uint64_t previousValue;
+    
+        uint64_t getValue(){
+            auto msg = can->last_msgs[messageID];
+            uint64_t* data = (uint64_t*)(msg.dat.data());
+            return getBitValue(*data, offset, size);
+        }
+};
+
+class SignalFilterer {
```

Comment:
Is it possible that these filter classes can be replaced by std library, e.g. std::greater, std::less, that might saves a lot of code.  (like what we did in void LogsWidget::setFilter())

Reply:
some of then probably could, but others probably couldn't ("increased value by...", "value between"). We could probably replace it with simple functions instead of classes though.